samples/msteams-chat-with-your-data/README.md

```diff
@@ -0,0 +1,92 @@
+
+This sample showcases a Teams app template to chat with your data via RAG (retrieval augmentation generation) through Azure OpenAI On Your Data and Azure AI Search.
+- The user needs to set up a datasource such as Azure AI Search, Azure Blob Storage, URL/web address, Azure Cosmos DB for MongoDB vCore, uploaded files, or Elasticsearch.
+- The user can then interact with the data through the Teams app template.
+  
+## Interacting with the app
+ ![bot-in-action](images/app.png)
+
+## Included Features
+* **Azure Open AI:** Azure OpenAI On Your Data leverages OpenAI's language models, like GPT-3.5 and GPT-4, along with RAG to provide a conversational interface for interacting with your domain data using the chat completion feature.
+* **Azure AI Search:** Utilizes Azure AI Search as a data cource for Azure OpenAI's chat completion feature.
+* **Azure Blob Storage:** The uploaded files are stored in a container in Azure Blob Storage.
+  
+## Setting up the app in Github Codespaces
+1. Click Open in GitHub Codespaces badge above to create a codespace for the sample app. Wait for the codespace to be setup, it may take a couple of minutes.
+2. Using the Teams Toolkit extension, sign in to your Microsoft 365 account and Azure account under ```ACCOUNTS```.
+3. [Set up your data source using Azure AI resources](#setting-up-your-data-source).
+4. [Populate the environment files](#populating-the-env-files).
+5. Press **Ctrl+Shift+D** to open the ```Run and Debug``` menu. Select ```Debug``` and press ```F5``` or click on the play button.
+6. Download the zip file ```appPackage/build/appPackage.local.zip``` and [sideload the app to Teams](#sideloading-the-app-to-teams).
+
+## Setting up the app locally
+### Prerequisites
+- [Python 3.11](https://www.python.org/downloads/)
+- [Node.js](https://nodejs.org/)
+- [Rust](https://www.rust-lang.org/tools/install)
+- [Poetry](https://python-poetry.org/docs/#installation)
+- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+- [Visual Studio Code](https://code.visualstudio.com/download)
+- [Teams Toolkit Extension ](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension)
+- [Python Extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+  
+1. Clone the repository
+   ```git clone https://github.com/t-mbindra/chat-with-your-data.git```
+2. Open the project folder(Microsoft-Teams-Samples/samples/msteams-chat-with-your-data) in VS Code.
+3.  Run
+   ```poetry install && poetry build```
+4. Using the Teams Toolkit extension, sign in to your Microsoft 365 account and Azure account under ```ACCOUNTS```.
+5. [Set up your data source using Azure AI resources](#setting-up-your-data-source).
+6. [Populate the environment files](#populating-the-env-files).
+7. Press **Ctrl+Shift+D** to open the ```Run and Debug``` menu. Press ```F5``` or click on the play button.
+
+## Deploying the app on Azure
```

Comment:
this step will be in last
https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/1322

Reply:
Done